core/trino-main/src/main/java/io/trino/execution/AddColumnTask.java

```diff
@@ -164,7 +168,7 @@ public ListenableFuture<Void> execute(
                     .map(Identifier::getValue)
                     .collect(toImmutableList());
 
-            ColumnMetadata columnMetadata = plannerContext.getMetadata().getColumnMetadata(session, tableHandle, columnHandles.get(columnName.getValue().toLowerCase(ENGLISH)));
```

Comment:
Why have we replaced this line ?

Reply:
Sorry my question was a bit misleading - Is there is any specific reason behind migrating the `Map<String, ColumnHandle>` to  `Map<String, ColumnMetadata>` - Now for supporting type coercion we need table properties so we required the `TableMetadata` - and we could derive the Map<String, ColumnMetadata>` - is that the reason or is there is reason for migrating the framework ?

Reply:
For most of the column specific operations we use this pattern - get the column handles and then get the column metadata based on the column handle - with this change - `AddColumnTask` looks a bit different

Reply:
That's the reason. It's just a side effect of getting `TableMetadata`.

Reply:
What if we could use `ConnectorMetadata#getTableProperties` ?

Reply:
https://github.com/trinodb/trino/pull/22499#discussion_r1658506881